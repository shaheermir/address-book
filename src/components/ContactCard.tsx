import React from 'react'
import { Link } from 'react-router-dom'
import { TContact } from '../features/types'

type TProps = {
  contact: TContact
}

export default function ContactCard({ contact }: TProps): JSX.Element {
  return (
    <div>
      <Link to={`/contact-details/${contact.id}`}>{contact.name.first}</Link>
    </div>
  )
}
