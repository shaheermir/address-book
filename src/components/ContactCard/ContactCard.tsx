import React from 'react'
import { Link } from 'react-router-dom'
import { TContact } from '../../features/types'
import styled from 'styled-components'

type TProps = {
  contact: TContact
}

const Container = styled(Link)`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 150px;
  text-decoration: none;
  text-align: center;
  margin-bottom: ${({ theme }) => theme.margins.bottom};
  margin-right: ${({ theme }) => theme.margins.bottom};
  color: ${({ theme }) => theme.colors.accent};
  padding-top: ${({ theme }) => theme.paddings.half};
  :hover {
    transform: scale(1.1);
    transition: transform 0.2s cubic-bezier(0.17, 0.67, 0.83, 0.67);
  }
`

const Name = styled.span``
const Avatar = styled.img`
  width: 70%;
  border-radius: 50%;
`

export function ContactCard({ contact }: TProps): JSX.Element {
  return (
    <Container to={`/contact-details/${contact.id}`}>
      <Avatar src={contact.picture.medium} />
      <Name>
        {contact.name.first} {contact.name.last}
      </Name>
    </Container>
  )
}
