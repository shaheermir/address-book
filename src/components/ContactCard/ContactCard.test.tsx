import React from 'react'
import { shallow } from 'enzyme'
import { ContactCard } from './ContactCard'
import contacts from '../ContactList/fixtures.json'

describe('ContactCard', () => {
  it('renders contact', () => {
    const link = shallow(<ContactCard contact={contacts[0]} />)
    expect(link).toMatchSnapshot()
  })
})
