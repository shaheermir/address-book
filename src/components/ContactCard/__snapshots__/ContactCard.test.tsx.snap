// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard renders contact 1`] = `
<Styled(Link)
  to="/contact-details/0-allan-nguyen"
>
  <styled.img
    src="https://randomuser.me/api/portraits/med/men/40.jpg"
  />
  <styled.span>
    Allan
     
    Nguyen
  </styled.span>
</Styled(Link)>
`;
